addon/components/file-dropzone/component.js

```diff
@@ -118,16 +118,30 @@ export default Ember.Component.extend({
   didDrop({ originalEvent: evt }) {
     // Testing support for dragging and dropping images
     // from other browser windows
-    let url = evt.dataTransfer.getData('text/uri-list');
-    let html = evt.dataTransfer.getData('text/html');
-    if (html) {
+    let supportsUrls, supportsHtml, html, url;
+
+    try {
+      html = evt.dataTransfer.getData('text/html');
+      supportsHtml = true;
```

Comment:
Sure! I understand where you're going with it, but I'm thinking something closer to moving the `let` to the top of the file and only checking the first time we get into `drop`.

```js
let supportsHtml = null;

// ... later
if (supportsHtml == null) {
  try {
    supportsHtml = true;
  } catch () {
    supportsHtml = false;
  }
}
```